lib/src/trie/proof_decode.rs

```diff
@@ -1607,6 +1612,8 @@ pub struct TrieNodeInfo<'a, T> {
     pub storage_value: StorageValue<'a>,
     /// Which children the node has.
     pub children: Children<'a, T>,
+    /// Entry in the proof
+    pub node_value: &'a [u8],
```

Comment:
That field is fundamentally wrong. The `TrieNodeInfo` struct gives some information about the trie, but that doesn't necessarily correspond to something found in the proof.

For example, if the proof shows that the trie contains the key `0xaaaa` with no children, then you can call `trie_node_info(0xaaaa2)` and a `Ok(TrieNodeInfo)` will be returned indicating that `0xaaaa2` doesn't exist (i.e. has no storage value and no children).
In that example, the `node_value` field doesn't make sense.